builtin/providers/aws/resource_aws_autoscaling_group.go

```diff
@@ -545,6 +557,25 @@ func getLBInstanceStates(g *autoscaling.Group, meta interface{}) (map[string]map
 	return lbInstanceStates, nil
 }
 
+func getInstanceSystemStatus(instance_id *string, meta interface{}) (string, error) {
+	ec2conn := meta.(*AWSClient).ec2conn
+
+	ids := []*string{instance_id}
+
+	input := &ec2.DescribeInstanceStatusInput{
+		InstanceIDs:         ids,
+		IncludeAllInstances: aws.Boolean(true),
+	}
+
+	output, err := ec2conn.DescribeInstanceStatus(input)
+	if err != nil {
+		return "", err
+	}
+
```

Comment:
Can we add a super-conservative guard here to avoid panics on weird responses?

Something like:

```go
if len(output.InstanceStatuses) != 1 || output.InstanceStatuses[0].SystemStatus == nil {
  return "", fmt.Errorf("Unexpected response while trying to describe instance status: %s", output)
}
```